backlight: always allow backlight to dim/wake

gudnimg opened the pull request:
This commit adds the ability for the firmware to dim and wake the backlight when LCD updates are disabled. Such as in the MMU error screen or when rendering full screen messages which typically disable the LCD updates to prevent the status screen from rendering.

Fixes #2777

Fixes an issue where the backlight may stay dim on MMU error screen even though the knob is pressed or rotated. Only happens when the Sound settings are set to Silent.

Change in memory:
Flash: -6 bytes
SRAM: +1 byte

gudnimg committed:
backlight: always allow backlight to dim/wake

This commit adds the ability for the firmware to dim and wake the
backlight when LCD updates are disabled. Such as in the MMU error screen
or when rendering full screen messages which typically
disable the LCD updates to prevent the status screen from rendering.

Fixes #2777

Change in memory:
Flash: -26 bytes
SRAM: +1 byte

leptun commented on Firmware/lcd.cpp:
```diff
@@ -776,6 +787,7 @@ void lcd_buttons_update(void)
 	if (lcd_buttons & EN_B) enc |= B10;
 	if (enc != lcd_encoder_bits)
 	{
+		lcd_backlight_wake_trigger = true; // flag event, knob rotated
```
I'm not sure this is entirely correct. I feel like this will make the backlight wake even if the encoder was moved only 1/4 of a tick of rotation. That might be enough to create random wakes of the backlight if the knob is between two encrot steps and oscillates between them due to noise/vibrations.

gudnimg replied:
That’s a good point. 👍

gudnimg replied:
Resolved with [8a926fa](https://github.com/prusa3d/Prusa-Firmware/pull/4010/commits/8a926fad103b1409701a092b98f5137ceff6ad5e)

leptun commented:
Besides the comment I wrote above, the PR looks fine to me.

gudnimg committed:
backlight: set rotation flag when certain amount of pulses are detected

We don't want to trigger a backlight wakeup randomly, for example
due to background noise.

leptun reviewed (approved):
Tested on MK3S 👍
